torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp

```diff
@@ -96,8 +119,22 @@ ParallelTypeBitmap avoidRedundantWrites(const TensorView* out_tv) {
     // If the axis is a broadcast domain and is parallelized by TID,
     // it is sufficient to use just one thread since the tensor is on
     // shared memory.
-    if (out_tv->getMemoryType() == MemoryType::Shared &&
-        out_tv_id->isBroadcast() && isParallelTypeThreadDim(pt)) {
+    if ((out_tv->getMemoryType() == MemoryType::Shared &&
+         out_tv_id->isBroadcast() && isParallelTypeThreadDim(pt)) ||
+        // Protect against global memory and is_reduction as we don't want to
```

Comment:
Took a while to remember what's going on here. Added a comment and filed an issue. #1671